packages/slice-machine/src/features/slices/sliceCards/SharedSliceViewCard.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  Box,
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Icon,
+  IconButton,
+  Text,
+  tokens,
+} from "@prismicio/editor-ui";
+import Link from "next/link";
+import type { FC } from "react";
+
+import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { Card, CardFooter, CardMedia, CardStatus } from "@src/components/Card";
+import {
+  countMissingScreenshots,
+  getScreenshotUrl,
+  getSharedSliceLibraryName,
+} from "@src/domain/Slice";
+import {
+  StatusBadge,
+  type StatusBadgeProps,
+} from "@src/features/changes/StatusBadge";
+import { SLICES_CONFIG } from "@src/features/slices/slicesConfig";
+import { AddPhotoAlternateIcon } from "@src/icons/AddPhotoAlternateIcon";
+
+type SharedSliceViewCardProps = {
+  action:
+    | { type: "menu"; onRemove: () => void; onRename: () => void }
+    | { type: "remove"; onRemove: () => void }
+    | ({ type: "status" } & Omit<StatusBadgeProps, "modelType">);
+  isDeletedSlice: boolean;
+  onUpdateScreenshot: (() => void) | undefined;
+  slice: ComponentUI;
+};
+
+export const SharedSliceViewCard: FC<SharedSliceViewCardProps> = ({
+  action,
+  isDeletedSlice,
+  onUpdateScreenshot,
+  slice,
+}) => {
+  const defaultVariation = ComponentUI.variation(slice);
+  const href = defaultVariation
+    ? SLICES_CONFIG.getBuilderPagePathname({
+        libraryName: slice.href,
+        sliceName: slice.model.name,
+        variationId: defaultVariation.id,
+      })
+    : undefined;
+  const src = defaultVariation
+    ? getScreenshotUrl(slice, defaultVariation)
+    : undefined;
+  const canUpdateScreenshot = !!onUpdateScreenshot && !isDeletedSlice;
+  return (
+    <Card
+      {...(href !== undefined && !isDeletedSlice
+        ? { component: Link, href, interactive: true }
+        : { interactive: false })}
+    >
+      {src !== undefined ? (
+        <CardMedia
+          overlay={
+            canUpdateScreenshot ? (
+              <Box alignItems="center" justifyContent="center">
+                <UpdateScreenshotButton onClick={onUpdateScreenshot} />
+              </Box>
+            ) : undefined
+          }
+          src={src}
+        />
+      ) : (
+        <CardMedia component="div">
+          <Box
+            alignItems="center"
+            flexDirection="column"
+            gap={8}
+            justifyContent="center"
+          >
+            <Text color="grey11" component="span">
+              No screenshot available
+            </Text>
+            {canUpdateScreenshot ? (
+              <UpdateScreenshotButton onClick={onUpdateScreenshot} />
+            ) : undefined}
+          </Box>
+        </CardMedia>
+      )}
+      <CardFooter
+        action={
```

Comment:
❓ I was talking about that for the action.

Here I feel we are passing down for example an action type `menu` then an action `onRemove` to construct the menu. But we could have exported a component with all that and prevent this component to be bigger by having a more composable approach, no? 

Since `action` is not used anywhere else it could be simpler, but let's discuss that